### ISSUE TITLE
Layout isn't used correctly in some cases

### ISSUE THREAD

**Issue by henri-tremblay** [asciidoctor/jekyll-asciidoc#187]
I've migrated from 1.0.1 to higher. 1.1.2 doesn't work and 1.1.1 has another issue.

I use jekyll 3.7.4. The original source code for this issue is [here](https://github.com/ehcache/ehcache.org-site).

 So `_config.xml` looks like this:

```yml
plugins:
  - jekyll-asciidoc
  - jekyll-sitemap


asciidoctor:
  # We need safe since we go fetch code examples in directories lower than where the adoc file is
  safe: 0
  attributes:
    - hardbreaks!
    - source-highlighter=prettify
    - pygments-css=style
    - icons=font
    - sourcedir36=_eh36
    - imagesdir=images
    - sectanchors
    - idprefix
    - idseparator=-

defaults:
  -
    values:
      layout: "default"
  -
    scope:
      path: "documentation/3.6"
      type: "pages"
    values:
      layout: "docs35_page"
```

So I would expect "docs35_page" to be selected when in the "documentaion/3.6" directory. But this is not happening.

The front matter  is each page is empty and is following by an include.

```
---
---
= Cache Event Listeners
ifndef::sourcedir36[]
include::common.adoc[]
endif::sourcedir36[]
```

The `common.adoc` file seems to contain nothing related.

```
---
---
ifndef::sourcedir36[]
:notBuildingForSite: true
ifdef::basebackend-html[:outfilesuffix: .html]
:source-highlighter: coderay
:stylesdir: css
:stylesheet: ehcache.css
:linkcss:
:icons: font
:iconfont-remote!:
:iconfont-name: font-awesome.min
:sourcedir36: ../../../../../
:imagesdir: images
:sectanchors:
:idprefix:
:idseparator: -
endif::sourcedir36[]
```

I have no clue why this is happening.

**Comment by mojavelinux** [comment]
I think it has to do with collection scopes. There's a similar issue reported here: #134. When this plugin was first written, collection scopes didn't exist (to my knowledge), so it may not fully support that use case.

If you know how to fix it, feel free to submit a test case and fix.

**Comment by henri-tremblay** [comment]
I have no idea how to fix. The strange thing is that it uses to work. So it's more a regression. Also, I'm surprised that the problem comes from asciidoc. In my head, the layout management was done by jekyll directly.

I will try to provide a test case.

**Comment by mojavelinux** [comment]
Oh, I see you were coming from the 1.x version of this plugin. That's a long time and many changes ago. The way the plugin works changed substantially from them, so we need to talk about the current logic is. It's hardly even worth comparing to the 1.x series (I barely even remember that code).

**Comment by mojavelinux** [comment]
The question is, why does this code not see the data from the scope?

https://github.com/asciidoctor/jekyll-asciidoc/blob/master/lib/jekyll-asciidoc/integrator.rb#L56-L101

Is it that the scope values are set later?

My assumption was that Jekyll would have already prepared the data map at this point. Maybe that's not the case.

**Comment by henri-tremblay** [comment]
Here is a reproduction of the problem: https://github.com/henri-tremblay/jekyllbug. Using 1.0.1, it works. Using anything above, it fails. Failing means http://127.0.0.1:4000/documentation/test isn't using the alternative layout.

**Comment by henri-tremblay** [comment]
I'm mentioning this code because that's the starting point where it stopped working.

**Comment by henri-tremblay** [comment]
Remember: Using the same jekyll version (and I tried 3.1.3 to 3.7.3) and just flipping the jekyll-asciidoc version makes it work. So I don't think it's coming from jekyll.

It is possible however that the value we are looking for is actually in 'page-layout'?

**Comment by mojavelinux** [comment]
1.x is so old that it's like a different plugin. So it really doesn't matter what it did before. The question is, what is the logic to make it work now?

My hypothesis is that Jekyll isn't passing the scoped values correctly, so the plugin never sees the layout value.

A demo project to reproduce would help. The project you linked to is empty (at least in my view). Even better would be a failing test.

The name of the variable we are looking for is layout. The name page-layout would only be used when set in the AsciiDoc document header. But then the page- prefix is stripped and it becomes the layout variable. But since you aren't using the AsciiDoc document header, that doesn't apply here.

**Comment by mojavelinux** [comment]
(This also could be an ordering issue. Whatever processes the scopes might be running after this plugin).

**Comment by mojavelinux** [comment]
> 1.x is so old that it's like a different plugin.

What I mean by that is that the architecture was completely changed to use Jekyll's lifecycle, so this plugin doesn't work in the same way it did in 1.x, and it's not going back.

**Comment by mojavelinux** [comment]
It's just very strange to me that a plugin has to worry about the front matter defaults at all (which is where the scoped variables get fed into). The plugin should just be able to refer to data and get all the resolved front matter, whether it's coming from _config.yml or the top of the file.

**Comment by mojavelinux** [comment]
Thanks. I'll run it and see what I find.

**Comment by henri-tremblay** [comment]
Thanks. I wondered for 3 seconds if it wasn't that that caused the problem. It's not.

**Comment by mojavelinux** [comment]
I see what the problem is. The keys in the data hash are lazy. So if we check `key? 'layout'`, the result is false, but if we check `['layout']`, then we get a value (effectively true). There's a missing testing for this, but your example provides me what I need to make it.

This could break the current rules for layout resolution. I'm still trying to sort that part out.

**Comment by mojavelinux** [comment]
I believe the linked PR solves the issue.

**Comment by mojavelinux** [comment]
I've released 2.1.1. Let me know if you have any trouble with it.
